pyproject.toml

```diff
@@ -1,9 +1,64 @@
+# -- Build system requirements (PEP 518) --
+
 [build-system]
 build-backend = 'setuptools.build_meta'
-requires = ['setuptools>=65.5.0', 'wheel>=0.33.6', 'cython>=0.29.13']
+requires = ['setuptools>=69.0.3', 'wheel>=0.33.6', 'cython>=3.0.0']
+
+# -- Dependency groups --
+[dependency-groups]
+build = [
+  'bump-my-version>=0.16.0',
+  'cython>=3.0.0',
+  'pip>=22.1.1',
+  'setuptools>=69.0.3',
+  'wheel>=0.33.6'
+]
+dev = [
+  {include-group = 'build'},
+  {include-group = 'docs'},
+  {include-group = 'lint'},
+  {include-group = 'test'},
+  {include-group = 'typing'}
+]
+docs = [
+  'esbonio>=0.16.0',
+  'jupytext>=1.14',
+  'matplotlib>=3.3',
+  'myst-parser>=4.0.0',
+  'pygments>=2.7.3',
+  'sphinx>=7.3.7',
+  'sphinx-rtd-theme>=3.0.1',
+  'sphinx-toolbox>=3.8.1'
+]
+lint = [
+  'pre-commit>=4.0.1',
+  'ruff>=0.8.0',
+  'tach>=0.16.0'
+]
+test = [
+  'atlas4py>=0.35',
```

Comment:
does it belong here?

Reply:
It is not really a gt4py dependency but an external package which can be used by gt4py users. In gt4py, it is only used in tests.